coders/heic.c

```diff
@@ -1319,17 +1319,49 @@ static MagickBooleanType WriteHEICImage(const ImageInfo *image_info,
             if (status == MagickFalse)
               break;
           }
-        option=GetImageOption(image_info,"heic:chroma");
-        if (option != (char *) NULL)
+      }
+#endif
+#if LIBHEIF_NUMERIC_VERSION >= 0x01080000
+    option=GetImageOption(image_info,"heic:chroma");
+    if (option != (char *) NULL)
+      {
+        error=heif_encoder_set_parameter(heif_encoder,"chroma",option);
+        status=IsHEIFSuccess(image,&error,exception);
+        if (status == MagickFalse)
+          break;
+      }
+#endif
+    options=heif_encoding_options_alloc();
+#if LIBHEIF_NUMERIC_VERSION >= 0x01100000
+    option=GetImageOption(image_info,"heic:chroma-downsampling-algorithm");
+    if (option != (char *) NULL)
+      {
+        if (LocaleCompare(option,"nearest-neighbor") == 0)
           {
-            error=heif_encoder_set_parameter(heif_encoder,"chroma",option);
-            status=IsHEIFSuccess(image,&error,exception);
-            if (status == MagickFalse)
-              break;
+            options->color_conversion_options.
+              preferred_chroma_downsampling_algorithm =
+                heif_chroma_downsampling_nearest_neighbor;
+            options->color_conversion_options.
+              only_use_preferred_chroma_algorithm = MagickTrue;
```

Comment:
I think we should use `1` here instead?

Reply:
I'm happy to use `1` here. I wasn't sure what the proper style was here. libheif uses C `true` and `false` which I know ImageMagick avoids.